app/actions/remote/reactions.ts

```diff
@@ -87,23 +88,28 @@ export const removeReaction = async (serverUrl: string, postId: string, emojiNam
         return {error};
     }
 };
-
 export const handleReactionToLatestPost = async (serverUrl: string, emojiName: string, add: boolean, rootId?: string) => {
     try {
-        const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const { database } = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
         let posts: PostModel[];
         if (rootId) {
             posts = await getRecentPostsInThread(database, rootId);
         } else {
             const channelId = await getCurrentChannelId(database);
             posts = await getRecentPostsInChannel(database, channelId);
         }
+        const latestRegularUserPost = posts.find(post => !isSystemMessage(post));
 
-        if (add) {
-            return addReaction(serverUrl, posts[0].id, emojiName);
+        if (latestRegularUserPost) {
+            if (add) {
+                return addReaction(serverUrl, latestRegularUserPost.id, emojiName);
+            }
+            return removeReaction(serverUrl, latestRegularUserPost.id, emojiName);
         }
-        return removeReaction(serverUrl, posts[0].id, emojiName);
+
+        return { error: 'No regular user posts found in the thread.' };
```

Comment:
To make this a bit shallower you can do this instead:

```
if (!latestRegularUserPost) {
    return { error: 'No regular user posts found in the thread.' };
}

if (add) {
...
```

Reply:
Is the error visible to the user? If so, it should be translatable